### PR TITLE
Test v2 L0 adapter using CTS

### PR DESCRIPTION
**Pull request by igchor**

### CODE REVIEW

File: test/conformance/CMakeLists.txt [hunk 1]
```diff
@@ -38,6 +41,10 @@ function(add_test_adapter name adapter)
     endif()
 
     set(TEST_ENV UR_ADAPTERS_FORCE_LOAD="$<TARGET_FILE:ur_${adapter}>")
+    if("${ARGN}" STREQUAL "v2")
+        list(APPEND TEST_ENV "UR_L0_USE_QUEUE_V2=1")
+        list(APPEND TEST_ENV "UR_L0_USE_IMMEDIATE_COMMANDLISTS=1")
```
Comment (pbalcer) [inline test/conformance/CMakeLists.txt:46]: why?

**Review by pbalcer — APPROVED** [review]
lgtm, but please rebase.

File: test/conformance/CMakeLists.txt (hunk 1 above)
Comment (igchor) [reply test/conformance/CMakeLists.txt:46]: Setting UR_L0_USE_IMMEDIATE_COMMANDLISTS is needed to make sure that the dispatcher will create v2_queue (not legacy queue). Alternatively we can modify the dispatcher to always use immediate cmd lists whenever UR_L0_USE_QUEUE_V2=1 is set.

But eventually, we will probably have separate test cases for immediate = 0 (once this path is implemented) anyway.

**Review by EwanC — APPROVED** [review]
command-buffer CTS file changes LGTM.

As an aside could somebody explain what the V2 L0 adapter is, and the plans for it?

**Comment by pbalcer** [comment]
> As an aside could somebody explain what the V2 L0 adapter is, and the plans for it?

We've decided that, to improve performance and maintainability of the L0 adapter, we need to make significant changes in the way queues are implemented. Given the accumulated complexity, the easiest way to do that is a full rewrite of that part (mostly queue and event components) of the code. But that's going to take a while, so all the fixes and features will land in the existing L0 adapter codebase, while we develop the V2 replacement. Once V2 is fully functional, we will switch over.

Right now our hope is to make the v2 stuff as minimally disruptive to other team as possible.

**Review by pbalcer — APPROVED** [review]

**Pull request merged**